ui/components/multichain/network-list-menu/network-list-menu.js

```diff
@@ -341,7 +341,6 @@ export const NetworkListMenu = ({ onClose }) => {
                                   selected={isCurrentNetwork}
                                   focus={isCurrentNetwork && !showSearch}
                                   onClick={() => {
-                                    dispatch(toggleNetworkMenu());
```

Comment:
Hmm, it's unclear how this relates to the problem. I know you mentioned in Slack that this had changed recently, but the previous version behaved the same way (the toggle happened first, see here: https://github.com/MetaMask/metamask-extension/blame/e1da0ef183279fcb5c4e2b64f7696533d3486ac0/ui/components/multichain/network-list-menu/network-list-menu.js#L344)

In both cases, the modal is closed before the switch completes, because we're not `await`iing the call to switch networks.

Reply:
reverted [here](05bf3655077d7d9641663f51af1258b56b2bfab4)